### PR TITLE
MEN-5452: test: device lifecycle backend integration tests for AWS IoT Core

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

**Comment by tranchitella** [comment]
Depends on: https://github.com/mendersoftware/iot-manager/pull/66

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --pr iot-manager/66 --fast

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --pr iot-manager/66 --fast

**Comment by tranchitella** [comment]
It also requires https://github.com/mendersoftware/mender-test-containers/pull/209

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --pr iot-manager/66 --fast

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --pr iot-manager/66 --fast

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --pr iot-manager/66 --fast

File: extra/release_tool.py [hunk 1]
```diff
@@ -2863,15 +2868,15 @@ def do_set_version_to(args):
...
-            os.path.join(integration_dir(), "extra", "mtls"), component, args.version,
-        )
-        set_component_version_to(
-            os.path.join(integration_dir(), "extra", "failover-testing"),
+            os.path.join(integration_dir(), "backend-tests", "docker"),
             component,
             args.version,
         )
         set_component_version_to(
-            os.path.join(integration_dir(), "extra", "mender-gateway"),
```
Comment (kacf) [inline extra/release_tool.py:2874]: Why remove this? Isn't it present anymore?

Comment (tranchitella) [reply extra/release_tool.py:2874]: it's a weird diff, I simply added backend-tests and sorted them to match in the two portions of code.

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --pr iot-manager/66 --fast

File: extra/release_tool.py [hunk 2]
```diff
@@ -2840,6 +2840,11 @@ def do_set_version_to(args):
         for assoc in component.associated_components_of_type("docker_image"):
             set_component_version_to(integration_dir(), assoc, args.version)
             # Update extra files used in integration tests
+            set_component_version_to(
+                os.path.join(integration_dir(), "backend-tests", "docker"),
+                assoc,
+                args.version,
+            )
```
Comment (kacf) [inline extra/release_tool.py:2847]: FYI, I only reviewed this part.

**Review by kacf — APPROVED** [review]

**Review by kjaskiewiczz — APPROVED** [review]
looks good

**Pull request merged**